gaia-media-controls.js

```diff
@@ -0,0 +1,291 @@
+;(function(define){'use strict';define(function(require,exports,module){
+
+/**
+ * Dependencies
+ */
+var Component = require('gaia-component');
+var MediaControlsImpl = require('./lib/media-controls-impl');
+
+var MediaControls = Component.register('gaia-media-controls', {
+  /**
+   * 'createdCallback' is called when the element is first created.
+   */
+  created: function() {
+    console.log('creating gaia-media-controls web component...');
+  },
+
+  attachTo: function(player) {
+
+    if (this.mediaPlayerImpl) {
+      throw new Error('A media player is already attached to the media controls component');
+    }
+
+    if (!this.shadowRoot) {
+      this.setupShadowRoot();
+    }
+    this._impl = new MediaControlsImpl(this, this.shadowRoot, player);
+  },
+
+  detachFrom: function() {
+    if (this.mediaPlayerImpl) {
+      this.mediaPlayerImpl.unload();
+      this.mediaPlayerImpl = null;
+    }
+  },
+
+  /*
+   * Expose testing helper functions
+   */
+  enableComponentTesting() {
+    if (this._impl) {
+      this._impl.enableComponentTesting();
+
+      var componentTestingHelper = {
+        triggerEvent:
+          this._impl.triggerEvent.bind(this._impl),
+        getElement:
+          this._impl.getElement.bind(this._impl),
+        disableComponentTesting:
+          this._impl.disableComponentTesting.bind(this._impl)
+      };
+    }
+
+    return componentTestingHelper;
+  },
+
+  template: `
+
+  <style>
+
+  @font-face {
+  	font-family: "gaia-icons";
+  	src: url("fonts/gaia-icons.ttf") format("truetype");
+  	font-weight: 500;
+  	font-style: normal;
+  }
+
+  [data-icon]:before {
+  	font-family: "gaia-icons";
+  	content: attr(data-icon);
+  	display: inline-block;
+  	font-weight: 500;
+  	font-style: normal;
+  	text-decoration: inherit;
+  	text-transform: none;
+  	text-rendering: optimizeLegibility;
+  	font-size: 30px;
+  	-webkit-font-smoothing: antialiased;
+  }
```

Comment:
Do you get this style for free when you load the gaia-icons stylesheet? It may not be required here, I think.